lib/ansible/modules/packaging/os/redhat_repository.py

```diff
@@ -0,0 +1,275 @@
+#!/usr/bin/python
+
+# Giovanni Sciortino (giovannibattistasciortino@gmail.com)
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+
+DOCUMENTATION = '''
+---
+module: redhat_repository
+short_description: Manage RHSM repositories using the C(subscription-manager) command
+description:
+    - Manage(List/Enable/Disable) RHSM repositories to the Red Hat Subscription Management entitlement platform using the C(subscription-manager) command
+version_added: "2.4"
+author: "Giovanni Sciortino (giovannibattistasciortino@gmail.com)"
+notes:
+    - In order to manage rhsm repositories the system must be already registered to rhsm manually or using the ansible module redhat_subscription.
+requirements:
+    - subscription-manager
+options:
+    list:
+        description:
+          - List all/enabled/disabled repositories
+        choices: [ "all", "enabled", "disabled" ]
+        required: False
+        default: None
+    enable:
+        description:
+          - id of repositories to enable
+          - To operate on several repositories this can accept a comma separated list or a list
+        required: False
+        default: None
+    disable:
+        description:
+          - id of repositories to disable
+          - To operate on several repositories this can accept a comma separated list or a list
+        required: False
+        default: None
+    disable_first:
+        description:
+          - If disable_first is true, first the repositories defined in 'disable' list are disabled and later the
+            repositories defined in 'enabled' are enabled
+          - If disable_first is false, first the repositories defined in 'enable' list are enabled and later the
+            repositories defined in 'disabled' are disabled
+        required: False
+        default: True
+
+'''
+
+EXAMPLES = '''
+- name: List all RHSM repositories.
+  redhat_repository:
+    list: all
+
+- name: List enabled RHSM repositories.
+  redhat_repository:
+    list: enabled
+
+- name: Enable a RHSM repository
+  redhat_subscription:
+    enable: rhel-7-server-rpms
```

Comment:
@albertomurillo thanks for your review
In the last commit I have added the list parameter as you suggest, now one(and only one) of the option list or state is required.
I have added in the documentation an example related to 'disable all task except one' based on your suggestion.

Note: For the option state I have used the values enabled/disabled instead of present/absent because "subscription-manager repos" enable/disable repositorities but it doesn't add/remove them from /etc/yum.repos.d/redhat.repo

I report below some examples with the new syntax:
```
- name: List all RHSM repositories.
  redhat_repository:
    list: all
  register: rhsm_repository

- name: List enabled RHSM repositories.
  redhat_repository:
    list: enabled
  register: enabled_rhsm_repository

- name: Enable a RHSM repository
  redhat_repository:
    name: rhel-7-server-rpms
    state: enabled

- name: Disable all RHSM repositories
  redhat_repository:
    name: '*'
    state: disabled

- name: Enable all repository starting with rhel-6-server
  redhat_repository:
    name: 'rhel-6-server*'
    state: enabled

- name: Disable all repositories except rhel-7-server-rpms
  redhat_repository:
    name: "{{ item }}"
    state: disabled
  with_items: "{{ rhsm_repository.repositories | map(attribute='id') | difference(['rhel-7-server-rpms']) }}"

```